ctp2_code/ui/interface/greatlibrary.cpp

```diff
@@ -1877,29 +1855,11 @@ void GreatLibrary::UpdateList( DATABASE database )
 		break;
 
 	case DATABASE_GOVERNMENTS:
-
-		for (index = 0; index < g_theGovernmentDB->NumRecords(); index++)
-		{
-			if(HIDE(g_theGovernmentDB, index)) continue;
-
-			Add_Item_To_Topics_List(g_theStringDB->GetNameStr(g_theGovernmentDB->GetName(
-						g_theGovernmentDB->m_alphaToIndex[ index ])), index);
-
-		}
-
+		AddTopics<GovernmentRecord>(g_theGovernmentDB);
 		break;
 
 	case DATABASE_TILE_IMPROVEMENTS:
-
-		for (index = 0; index < g_theTerrainImprovementDB->NumRecords(); index++)
```

Comment:
Of course make this local here as well.

Then m_maxIndex = index; can also go at the bottom of this function. Actually, the Windows debug version does not like it, since index may be uninitialized. Actually m_maxIndex is not used elsewhere. So remove it also from the class definition.

Reply:
Done